lib/model/message_list.dart

```diff
@@ -341,6 +341,38 @@ class MessageListView with ChangeNotifier, _MessageSequence {
     notifyListeners();
   }
 
+  void maybeUpdateMessageFlags(UpdateMessageFlagsEvent event) {
+    final List<Message> messagesToProcess;
+    if (event.all) {
+      messagesToProcess = messages;
+    } else {
+      messagesToProcess = [];
+      for (final messageId in event.messages) {
+        // TODO(perf) could be more efficient?
+        final index = _findMessageWithId(messageId);
```

Comment:
I think this algorithm is fine. For marking $k$ messages, with $n$ in the message list, it will take $k \log_2 n$ steps to find them all (or respectively determine they're not present). Even if there are thousands of messages in this message list, that's no more than 10-15 steps each (and 15 would be really a *lot* of messages, over 32k of them).

I think the most efficient possible with these data structures is probably $k \log_2 (n/k)$, and the complication doesn't seem worth it.

We could make it linear in $k$ with more data structures. But that can be out of scope here.